src/js/components/DataTable/Body.js

```diff
@@ -85,67 +89,101 @@ const Body = forwardRef(
+                              childExpand.filter(s => s !== index),
+                            );
+                          } else {
+                            setChildExpand([...childExpand, index]);
+                          }
                         }}
                       />
-                    </TableCell>
-                  )}
-                  {columns.map(column => (
-                    <Cell
-                      key={column.property}
-                      background={column.pin ? pinnedBackground : background}
-                      border={border}
-                      context="body"
-                      column={column}
-                      datum={datum}
-                      index={index}
-                      pad={pad}
-                      primaryProperty={primaryProperty}
-                      rowProp={rowProps && rowProps[primaryValue]}
-                      scope={
-                        column.primary || column.property === primaryProperty
-                          ? 'row'
-                          : undefined
-                      }
-                    />
-                  ))}
-                </StyledDataTableRow>
+                    )}
+                    {(selected || onSelect) && (
+                      <TableCell background={background}>
+                        <CheckBox
+                          a11yTitle={`${
+                            isSelected ? 'unselect' : 'select'
+                          } ${primaryValue}`}
+                          checked={isSelected}
+                          disabled={!onSelect}
+                          onChange={() => {
+                            if (isSelected) {
+                              onSelect(
+                                selected.filter(s => s !== primaryValue),
+                              );
+                            } else onSelect([...selected, primaryValue]);
+                          }}
+                        />
+                      </TableCell>
+                    )}
+
+                    {columns.map(column => (
+                      <Cell
+                        key={column.property}
+                        background={column.pin ? pinnedBackground : background}
+                        border={border}
+                        context="body"
+                        column={column}
+                        datum={datum}
+                        index={index}
+                        pad={pad}
+                        primaryProperty={primaryProperty}
+                        rowProp={rowProps && rowProps[primaryValue]}
+                        scope={
+                          column.primary || column.property === primaryProperty
+                            ? 'row'
+                            : undefined
+                        }
+                      />
+                    ))}
+                  </StyledDataTableRow>
+                  <StyledDataTableRow key={`${index.toString()}_expand`}>
```

Comment:
We shouldn't be adding an extra table row unless we actually have a child.

Reply:
Addressed this.